### PR TITLE
Sources - ClickUp

### LINKED ISSUE #3972: [TRIGGER] ClickUp sources
**App**
ClickUp

**Sources**
New Task
Updated Task
New Folder
New List

Webhooks available, see API docs at https://jsapi.apiary.io/apis/clickup20/introduction/faq.html
Updated Task requested by user at https://github.com/PipedreamHQ/pipedream/issues/3950

### PR DESCRIPTION
**Pull request by vellames**
Closes #3972 

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4101"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/clickup/sources/new-folder/new-folder.mjs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  description: "Emit new event when a new folder is created",
+  version: "0.0.1",
+  dedupe: "unique",
+  type: "source",
+  methods: {
+    ...common.methods,
+    _getMeta({ folder_id: folderId }) {
+      return {
+        id: folderId,
+        summary: folderId,
```
Comment (feyzullah) [inline components/clickup/sources/new-folder/new-folder.mjs:16]: Giving a summary more than `folderId` might be better for user experience.

File: components/clickup/sources/new-list/new-list.mjs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  description: "Emit new event when a new list is created",
+  version: "0.0.1",
+  dedupe: "unique",
+  type: "source",
+  methods: {
+    ...common.methods,
+    _getMeta({ list_id: listId }) {
+      return {
+        id: listId,
+        summary: listId,
```
Comment (feyzullah) [inline components/clickup/sources/new-list/new-list.mjs:16]: Again, giving a summary more than `listId` might be better for user experience.

File: components/clickup/sources/new-task/new-task.mjs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  description: "Emit new event when a new task is created",
+  version: "0.0.1",
+  dedupe: "unique",
+  type: "source",
+  methods: {
+    ...common.methods,
+    _getMeta({ task_id: taskId }) {
+      return {
+        id: taskId,
+        summary: taskId,
```
Comment (feyzullah) [inline components/clickup/sources/new-task/new-task.mjs:16]: Again, giving a summary more than `taskId` might be better for user experience.

**Review by feyzullah — APPROVED** [review]
LGTM. Added a few comments for event summaries.

**Comment by vellames** [comment]
The webhook payload don't provide more informations :(

File: components/clickup/sources/new-task/new-task.mjs (hunk 3 above)
Comment (vellames) [reply components/clickup/sources/new-task/new-task.mjs:16]: The webhook payload don't provide more informations :(

File: components/clickup/sources/new-list/new-list.mjs (hunk 2 above)
Comment (vellames) [reply components/clickup/sources/new-list/new-list.mjs:16]: The webhook payload doesn't provide more information :(

File: components/clickup/sources/new-folder/new-folder.mjs (hunk 1 above)
Comment (vellames) [reply components/clickup/sources/new-folder/new-folder.mjs:16]: The webhook payload doesn't provide more information :(

**Comment by DilanAthukorala** [comment]
@vellames  In "New Folder" source **folder_id** is not displayed for folders created through templates 
image

**Comment by vellames** [comment]
@DilanAthukorala Could you check again, please?

**Comment by DilanAthukorala** [comment]
Verification completed, This is ready for release.

**Review by lcaresia — APPROVED** [review]
LGTM!

**Pull request merged**